src/platforms/ctxlink/WiFi_Server.c

```diff
@@ -0,0 +1,1511 @@
+	}
+	if (uart_debug_client_socket != SOCK_ERR_INVALID) {
+		close(uart_debug_client_socket);
+		uart_debug_client_connected = false;
+		uart_debug_client_socket = SOCK_ERR_INVALID;
+	}
+	if (swo_trace_client_socket != SOCK_ERR_INVALID) {
+		close(swo_trace_client_socket);
+		swo_trace_client_connected = false;
+		swo_trace_client_socket = SOCK_ERR_INVALID;
+	}
+	//
+	// Servers
+	//
+	if (gdb_server_socket != SOCK_ERR_INVALID) {
+		gdb_tcp_server_state = sm_idle;
+		gdb_server_is_running = false;
+		close(gdb_server_socket);
+		gdb_server_socket = SOCK_ERR_INVALID;
+	}
+	if (uart_debug_server_socket != SOCK_ERR_INVALID) {
+		uart_debug_tcp_server_state = sm_idle;
+		uart_debug_server_is_running = false;
+		close(uart_debug_server_socket);
+		uart_debug_server_socket = SOCK_ERR_INVALID;
+	}
+	if (swo_trace_server_socket != SOCK_ERR_INVALID) {
+		swo_trace_tcp_serverstate = sm_idle;
+		swo_trace_server_is_running = false;
+		close(swo_trace_server_socket);
+		swo_trace_server_socket = SOCK_ERR_INVALID;
+	}
+}
+
+void gdb_tcp_server(void)
+{
+	switch (gdb_tcp_server_state) {
+	case sm_idle:
+		break; // Startup and testing do nothing state
+	case sm_home: {
+		//
+		// Allocate a socket for this server to listen and accept connections on
+		//
+		gdb_server_socket = socket(AF_INET, SOCK_STREAM, 0);
+		if (gdb_server_socket < SOCK_ERR_NO_ERROR)
+			return;
+		//
+		// Bind the socket
+		//
+		gdb_addr.sin_addr.s_addr = 0;
+		gdb_addr.sin_family = AF_INET;
+		gdb_addr.sin_port = _htons(GDB_SERVER_PORT);
+		const int8_t result = bind(gdb_server_socket, (struct sockaddr *)&gdb_addr, sizeof(gdb_addr));
+		if (result != SOCK_ERR_NO_ERROR)
+			return;
+		gdb_tcp_server_state = sm_listening;
+		break;
+	}
+
+	case sm_listening:
+		//
+		// No need to perform any flush.
+		// TCP data in TX FIFO will automatically transmit itself after it accumulates for a while.
+		// If you want to decrease latency (at the expense of wasting network bandwidth on TCP overhead),
+		// perform and explicit flush via the TCPFlush() API.
+		break;
+
+	case sm_closing:
+		// Close the socket connection.
+		close(gdb_server_socket);
+		gdb_server_socket = SOCK_ERR_INVALID;
+		gdb_tcp_server_state = sm_idle;
+		break;
+	}
+}
+
+void data_tcp_server(void)
+{
+	//
+	// UART Server
+	//
+	switch (uart_debug_tcp_server_state) {
+	case sm_idle: {
+		break; // Startup and testing do nothing state
+	}
+	case sm_home: {
+		//
+		// Allocate a socket for this server to listen and accept connections on
+		//
+		uart_debug_server_socket = socket(AF_INET, SOCK_STREAM, 0);
+		if (uart_debug_server_socket < SOCK_ERR_NO_ERROR) {
+			return;
+		}
+		//
+		// Bind the socket
+		//
+		uart_debug_addr.sin_addr.s_addr = 0;
+		uart_debug_addr.sin_family = AF_INET;
+		uart_debug_addr.sin_port = _htons(UART_DEBUG_SERVER_PORT);
+		int8_t result = bind(uart_debug_server_socket, (struct sockaddr *)&uart_debug_addr, sizeof(uart_debug_addr));
+		if (result != SOCK_ERR_NO_ERROR) {
+			return;
+		}
```

Comment:
This and the `if` after the `bind` call below still have some extra braces to be removed. Both `bind()` calls look like they could do with the result variable being made `const`.

Reply:
Scanned the whole file, hopefully I caught them all.

Reply:
It looks like it 👍🏼